### DISCUSSION TITLE
Using stock / container locations mapped to drawer ID's?

### DISCUSSION THREAD

**Discussion by StormTrooper** [inventree/InvenTree#5610] [Q&A]
I'm using inventree as a hobbiest and my collection of components is growing rapidly.  I've got about 200 bin drawers with multiple components in each drawer.  About 1000 components. 

Getting tired of trying to sort components alphabetically, otherwise I'm spending ages looking through all the bins for the compents, so was thinking of trying to use stock locations and some kind of drawer ID to store components.   The bins are labelled with the component type/value but keeping this up to date/sorted is a nightmare.

Looking at using a drawer ID system, so for example  all my drawers will be labelled BIN1 to BIN200, and then in Inventree on each compnent I assign it a BIN ID.  (Multiple components will be stored in each bin)

The only downside I see to this is when I receive the component and check it in on the purchase order I need to specify the location.   I don't want to have to look this up when checking in each purchase order. Can this value be filled in based on which  drawer the component is in?  (The components will only be in a specfic drawer and not split between multiple drawers)

Or is there some other way of doing this I'm overlooking?

**Comment by SchrodingersGat** [comment]
If I understand what you are asking for,it might be a good target for a simple plugin. 

When a line is added to a purchase order, look to see where that part has located stock. If in a single location, fill in the "receive location" for the purchase order line item accordingly.

**Reply by wolflu05** [reply]
I actually like the idea too. We might could also implement this in the new frontend, maybe via a simple filter button which is on by default, on the location select that only shows locations this part has stock in. If there is only one location, choose that. I'm not sure how that would work for large businesses, but for small size hobby instances that store most of the parts only on one location, this will save a lot of time.

**Comment by StormTrooper** [comment]
Thanks, will look at a plugin. I like the idea suggested above - default to choosing the location is one is already set.  This can be a toggle on/off option in the menu if you want/don't want to use it?

**Comment by StormTrooper** [comment]
Had a read through plugins but unfortunately this is way over my head.

**Reply by lippoliv** [reply]
If you're unfamiliar with coding it will get hard. 

But if you have some skills here, the [PanelMixin](https://docs.inventree.org/en/latest/extend/plugins/panel/) would be a good starting point. For every purchase order you can have a menu entry where per ordered item the locations are shown.

Theoretically it can be done in one single python file since Panels are not too hard. The setup would be quiet easy using [Devcontainers](https://docs.inventree.org/en/latest/develop/devcontainer/), they are prepared very well and it's described how you can easily start [developing plugins](https://docs.inventree.org/en/latest/develop/devcontainer/#plugin-development).

**Comment by StormTrooper** [comment]
Sounds promising. I know some python and can hack my way around. The code might not be pretty but it could work :)
